<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QtWidgetsApplication1Class</class>
 <widget class="QMainWindow" name="QtWidgetsApplication1Class">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>732</width>
    <height>547</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtWidgetsApplication1</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="openFileBtn">
    <property name="geometry">
     <rect>
      <x>310</x>
      <y>10</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>開啟圖檔</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_Image">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>381</width>
      <height>381</height>
     </rect>
    </property>
    <property name="text">
     <string>TextLabel</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_ROI">
    <property name="geometry">
     <rect>
      <x>540</x>
      <y>10</y>
      <width>181</width>
      <height>121</height>
     </rect>
    </property>
    <property name="title">
     <string>ROI</string>
    </property>
    <widget class="QLineEdit" name="lineEdit_X1">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>20</y>
       <width>41</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_X2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>20</y>
       <width>41</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_Y1">
     <property name="geometry">
      <rect>
       <x>50</x>
       <y>50</y>
       <width>41</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="lineEdit_Y2">
     <property name="geometry">
      <rect>
       <x>130</x>
       <y>50</y>
       <width>41</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_X">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>20</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X :</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_Y">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>50</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y :</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_Xtilde">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>20</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>~</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_Ytilde">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>50</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="font">
      <font>
       <family>Arial</family>
       <pointsize>14</pointsize>
      </font>
     </property>
     <property name="text">
      <string>~</string>
     </property>
    </widget>
    <widget class="QPushButton" name="roiBtn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>80</y>
       <width>161</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>ROI</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>130</y>
      <width>211</width>
      <height>71</height>
     </rect>
    </property>
    <property name="title">
     <string>二值化</string>
    </property>
    <widget class="QSlider" name="thresholdSlider">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>30</y>
       <width>181</width>
       <height>22</height>
      </rect>
     </property>
     <property name="maximum">
      <number>255</number>
     </property>
     <property name="singleStep">
      <number>1</number>
     </property>
     <property name="pageStep">
      <number>10</number>
     </property>
     <property name="value">
      <number>0</number>
     </property>
     <property name="sliderPosition">
      <number>0</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>281</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Arial</family>
      <pointsize>26</pointsize>
     </font>
    </property>
    <property name="text">
     <string>Original Image</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="groupBox_2">
    <property name="geometry">
     <rect>
      <x>500</x>
      <y>210</y>
      <width>111</width>
      <height>80</height>
     </rect>
    </property>
    <property name="title">
     <string>直方圖</string>
    </property>
    <widget class="QPushButton" name="histogramBtn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>顯示直方圖</string>
     </property>
    </widget>
    <widget class="QPushButton" name="equalHistBtn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>直方圖等化</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_3">
    <property name="geometry">
     <rect>
      <x>610</x>
      <y>210</y>
      <width>111</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>色彩空間轉換</string>
    </property>
    <widget class="QPushButton" name="grayButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>Gray</string>
     </property>
    </widget>
    <widget class="QPushButton" name="hsvButton">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>HSV</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_4">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>10</y>
      <width>111</width>
      <height>81</height>
     </rect>
    </property>
    <property name="title">
     <string>濾波器</string>
    </property>
    <widget class="QPushButton" name="gaussianBlur_Btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>GaussianBlur</string>
     </property>
    </widget>
    <widget class="QPushButton" name="medianBlur_Btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>50</y>
       <width>93</width>
       <height>28</height>
      </rect>
     </property>
     <property name="text">
      <string>MedianBlur</string>
     </property>
    </widget>
   </widget>
   <widget class="QGroupBox" name="groupBox_5">
    <property name="geometry">
     <rect>
      <x>420</x>
      <y>300</y>
      <width>301</width>
      <height>181</height>
     </rect>
    </property>
    <property name="title">
     <string>幾何轉換</string>
    </property>
    <widget class="QPushButton" name="warpAffine_Btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>20</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>平移</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="warpAffine_X_text">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>30</y>
       <width>41</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLineEdit" name="warpAffine_Y_text">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>30</y>
       <width>41</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="label_X_3">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>10</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>水平</string>
     </property>
    </widget>
    <widget class="QPushButton" name="rotate_Btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>60</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>旋轉</string>
     </property>
    </widget>
    <widget class="QLineEdit" name="rotate_text">
     <property name="geometry">
      <rect>
       <x>100</x>
       <y>60</y>
       <width>41</width>
       <height>22</height>
      </rect>
     </property>
    </widget>
    <widget class="QLabel" name="rotate_label">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>60</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>度</string>
     </property>
    </widget>
    <widget class="QPushButton" name="affineTransform_Btn">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>100</y>
       <width>81</width>
       <height>31</height>
      </rect>
     </property>
     <property name="text">
      <string>仿射轉換</string>
     </property>
    </widget>
    <widget class="QLabel" name="affineTransform_label_X">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>110</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>X </string>
     </property>
    </widget>
    <widget class="QLabel" name="affineTransform_label_Y">
     <property name="geometry">
      <rect>
       <x>280</x>
       <y>140</y>
       <width>21</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>Y</string>
     </property>
    </widget>
    <widget class="QLabel" name="affineTransform_label4">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>140</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>輸入倍率</string>
     </property>
    </widget>
    <widget class="QGroupBox" name="groupBox_6">
     <property name="geometry">
      <rect>
       <x>90</x>
       <y>90</y>
       <width>61</width>
       <height>80</height>
      </rect>
     </property>
     <property name="title">
      <string>原點</string>
     </property>
     <widget class="QLineEdit" name="affineTransform_text_1">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="affineTransform_text_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_7">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>90</y>
       <width>61</width>
       <height>80</height>
      </rect>
     </property>
     <property name="title">
      <string>座標1</string>
     </property>
     <widget class="QLineEdit" name="affineTransform_text_3">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="affineTransform_text_4">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QGroupBox" name="groupBox_8">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>90</y>
       <width>61</width>
       <height>80</height>
      </rect>
     </property>
     <property name="title">
      <string>座標2</string>
     </property>
     <widget class="QLineEdit" name="affineTransform_text_5">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="affineTransform_text_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>50</y>
        <width>41</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QLabel" name="affineTransform_label5">
     <property name="geometry">
      <rect>
       <x>20</x>
       <y>160</y>
       <width>61</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>0~1之間</string>
     </property>
    </widget>
    <widget class="QPushButton" name="perspectiveTransform_Btn">
     <property name="geometry">
      <rect>
       <x>210</x>
       <y>20</y>
       <width>81</width>
       <height>61</height>
      </rect>
     </property>
     <property name="text">
      <string>透視投影
轉換</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_X_2">
     <property name="geometry">
      <rect>
       <x>150</x>
       <y>10</y>
       <width>31</width>
       <height>16</height>
      </rect>
     </property>
     <property name="text">
      <string>垂直</string>
     </property>
    </widget>
   </widget>
   <widget class="QLabel" name="image_size_label_width">
    <property name="geometry">
     <rect>
      <x>300</x>
      <y>430</y>
      <width>31</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>W</string>
    </property>
   </widget>
   <widget class="QLabel" name="image_size_label_2">
    <property name="geometry">
     <rect>
      <x>330</x>
      <y>430</y>
      <width>16</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>*</string>
    </property>
   </widget>
   <widget class="QLabel" name="image_size_label_height">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>430</y>
      <width>41</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>H</string>
    </property>
   </widget>
   <widget class="QPushButton" name="openCapBtn">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>440</y>
      <width>93</width>
      <height>28</height>
     </rect>
    </property>
    <property name="text">
     <string>開啟視訊</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>732</width>
     <height>25</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="qtwidgetsapplication1.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>openFileBtn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>openFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>786</x>
     <y>72</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>roiBtn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>ROI()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>540</x>
     <y>572</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>histogramBtn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>histogram()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>796</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>thresholdSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>Qthreshold()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>529</x>
     <y>549</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>equalHistBtn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>QequalizeHist()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>566</x>
     <y>502</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>grayButton</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>change2Gray()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>576</x>
     <y>462</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>hsvButton</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>change2HSV()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>576</x>
     <y>492</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>gaussianBlur_Btn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>QgaussianBlur()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>786</x>
     <y>413</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>medianBlur_Btn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>QmedianBlur()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>786</x>
     <y>443</y>
    </hint>
    <hint type="destinationlabel">
     <x>426</x>
     <y>313</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>warpAffine_Btn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>QwarpAffine()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>460</x>
     <y>410</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotate_Btn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>Qrotate()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>400</x>
     <y>400</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>affineTransform_Btn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>QaffineTransform()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>450</x>
     <y>395</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>260</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>perspectiveTransform_Btn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>QperspectiveTransform()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>780</x>
     <y>300</y>
    </hint>
    <hint type="destinationlabel">
     <x>424</x>
     <y>340</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openCapBtn</sender>
   <signal>clicked()</signal>
   <receiver>QtWidgetsApplication1Class</receiver>
   <slot>openCap()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>156</x>
     <y>502</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>273</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>openFile()</slot>
  <slot>ROI()</slot>
  <slot>histogram()</slot>
  <slot>Qthreshold()</slot>
  <slot>QequalizeHist()</slot>
  <slot>change2Gray()</slot>
  <slot>change2HSV()</slot>
  <slot>QgaussianBlur()</slot>
  <slot>QmedianBlur()</slot>
  <slot>QwarpAffine()</slot>
  <slot>Qrotate()</slot>
  <slot>QaffineTransform()</slot>
  <slot>QperspectiveTransform()</slot>
  <slot>openCap()</slot>
 </slots>
</ui>
